dll/win32/syssetup/wizard.c

```diff
@@ -390,89 +391,152 @@ AckPageDlgProc(HWND hwndDlg,
     return FALSE;
 }
 
+static const WCHAR s_szProductOptions[] = L"SYSTEM\\CurrentControlSet\\Control\\ProductOptions";
+static const WCHAR s_szRosVersion[] = L"SYSTEM\\CurrentControlSet\\Control\\ReactOS\\Settings\\Version";
+static const WCHAR s_szWindowsNT[] = L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion";
+static const WCHAR s_szControlWindows[] = L"SYSTEM\\CurrentControlSet\\Control\\Windows";
+static const WCHAR s_szCurrentBuildNumber[] = L"CurrentBuildNumber";
+static const WCHAR s_szCurrentVersion[] = L"CurrentVersion";
+
+typedef struct PRODUCT_OPTION_DATA
+{
+    LPCWSTR ProductSuite;
+    LPCWSTR ProductType;
+    DWORD ReportAsWorkstation;
+    LPCWSTR CurrentVersion;
+    LPCWSTR CurrentBuildNumber;
+    DWORD CSDVersion;
+} PRODUCT_OPTION_DATA, *LPPRODUCT_OPTION_DATA;
+
+static const PRODUCT_OPTION_DATA s_ProductOptionData[] =
+{
+    {
+        L"Terminal Server\0",
+        L"ServerNT",
+        0,
+        L"5.2",
+        L"3790",
+        0x200
+    },
+    {
+        L"\0",
+        L"WinNT",
+        1,
+        L"5.1",
+        L"2600",
+        0x300
+    }
+};
+
 static BOOL
 DoWriteProductOption(PRODUCT_OPTION nOption)
 {
-    static const WCHAR s_szProductOptions[] = L"SYSTEM\\CurrentControlSet\\Control\\ProductOptions";
-    static const WCHAR s_szRosVersion[] = L"SYSTEM\\CurrentControlSet\\Control\\ReactOS\\Settings\\Version";
     HKEY hKey;
     LONG error;
-    LPCWSTR pData;
-    DWORD cbData, dwValue;
+    LPCWSTR pszData;
+    DWORD dwValue, cbData;
+    const PRODUCT_OPTION_DATA *pData = &s_ProductOptionData[nOption];
+    ASSERT(0 <= nOption && nOption < _countof(s_ProductOptionData));
 
-    error = RegOpenKeyExW(HKEY_LOCAL_MACHINE, s_szProductOptions, 0, KEY_WRITE, &hKey);
-    if (error)
-        return FALSE;
-
-    switch (nOption)
+    do
```

Comment:
Why not just use a common pattern `goto Error;`? :)

Reply:
OK.